pretyping/printingFlags.mli

```diff
@@ -0,0 +1,85 @@
+(************************************************************************)
+(*         *      The Rocq Prover / The Rocq Development Team           *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+module Detype : sig
+  type t = {
+    raw : bool;
```

Comment:
Future work (in another PR): remove this `raw` field (and the one for Extern) and instead have `current` override the other fields according to the current value of raw_print, and add combinator `make_raw : t -> t` to locally do raw printing.
This will require adding some new fields since some places only check `raw` instead of eg `raw && notations`.